shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -695,6 +695,11 @@ - (BOOL)createShell:(NSString*)entrypoint
   self.initialRoute = initialRoute;
 
   auto settings = [_dartProject.get() settings];
+  if (settings.route.empty() == false) {
+    self.initialRoute = [NSString stringWithCString:settings.route.c_str()
+                                           encoding:[NSString defaultCStringEncoding]];
+  }
```

Comment:
I think `defaultCStringEncoding` usage is discouraged, let's just accept utf8 here.
Also, don't set the `initialRoute` twice.

```suggestion
  auto settings = [_dartProject.get() settings];
  if (initialRoute != nil) {
    self.initialRoute = initialRoute;
  } else if (settings.route.empty() == false) {
    self.initialRoute = [NSString stringWithCString:settings.route.c_str()
                                           encoding:NSUTF8StringEncoding];
  }
```